sdks/client-go/pkg/models/strategies.go

```diff
@@ -0,0 +1,262 @@
+				continue
+			}
+			logger.Tracef("Didn't match attribute strategy (%s:%s = %v) for country: %v\n", sa.ID, sa.FieldName, sa.Values, clientContext.Country)
+			return false
+
+		// Match by device type:
+		case strategyFieldNameDevice:
+			if len(clientContext.Device) > 0 && sa.matchConditional(sa.Values, string(clientContext.Device)) {
+				continue
+			}
+			logger.Tracef("Didn't match attribute strategy (%s:%s = %v) for device: %v\n", sa.ID, sa.FieldName, sa.Values, clientContext.Device)
+			return false
+
+		// Match by platform:
+		case strategyFieldNamePlatform:
+			if len(clientContext.Platform) > 0 && sa.matchConditional(sa.Values, string(clientContext.Platform)) {
+				continue
+			}
+			logger.Tracef("Didn't match attribute strategy (%s:%s = %v) for platform: %v\n", sa.ID, sa.FieldName, sa.Values, clientContext.Platform)
+			return false
+
+		// Match by version:
+		case strategyFieldNameVersion:
+			logger.Trace("Trying version")
+			if len(clientContext.Version) > 0 && sa.matchConditional(sa.Values, string(clientContext.Version)) {
+				continue
+			}
+			logger.Tracef("Didn't match attribute strategy (%s:%s = %v) for version: %v\n", sa.ID, sa.FieldName, sa.Values, clientContext.Version)
+			return false
+
+		// Some other (unsupported) field:
+		default:
+			logger.Infof("Unsupported strategy field (%s)", sa.FieldName)
+			return false
+		}
+	}
+
+	return true
+}
+
+// matchConditional checks the given string against the given slice of strings with the attribute's conditional logic:
+func (sa *StrategyAttribute) matchConditional(slice []string, contains string) bool {
+
+	// Handle the different conditionals available to us:
+	switch {
+
+	// If our value is found in the slice then we match:
+	case sa.Conditional == strategyConditionalEquals:
+		for _, value := range slice {
+			if value == contains {
+				return true
+			}
+		}
+		return false
+
+	// If our value is found in the slice then we do NOT match:
+	case sa.Conditional == strategyConditionalNotEquals:
+		for _, value := range slice {
+			if value == contains {
+				return false
+			}
+		}
+		return true
+
+	// If our value <= any value in the slice then we do NOT match:
+	case sa.Conditional == strategyConditionalGreater && sa.Type == strategyTypeSemanticVersion:
+		for _, value := range slice {
+			if version.Compare(contains, value, "<=") {
+				return false
+			}
+		}
+		return true
+
+	// // If our value <= any value in the slice then we do NOT match:
+	// case sa.Conditional == strategyConditionalGreater && sa.Type != strategyTypeSemanticVersion:
+	// 	for _, value := range slice {
+	// 		if contains <= value {
+	// 			return false
+	// 		}
+	// 	}
+	// 	return true
+
+	// If our value < any value in the slice then we do NOT match:
+	case sa.Conditional == strategyConditionalGreaterEquals && sa.Type == strategyTypeSemanticVersion:
+		for _, value := range slice {
+			if version.Compare(contains, value, "<") {
```

Comment:
is this a string comparison? because 1.9.0 > 1.10.0 in a string?

Reply:
No! I'm using a semver comparison library for this logic. I have unit tests which prove that 1.9.0 is not greater than 1.10.0 as well.